src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs

```diff
@@ -932,14 +910,43 @@ protected override void EndProcessing()
             {
                 // Load the source if they want to load from a file
                 if (String.Equals(ParameterSetName, "FromPath", StringComparison.OrdinalIgnoreCase) ||
-                    String.Equals(ParameterSetName, "FromLiteralPath", StringComparison.OrdinalIgnoreCase))
+                    String.Equals(ParameterSetName, "FromLiteralPath", StringComparison.OrdinalIgnoreCase)
+                   )
                 {
-                    this.sourceCode = "";
-                    foreach (string file in paths)
+                    if (paths.Length == 1)
+                    {
+                        sourceCode = File.ReadAllText(paths[0]);
+                    }
+                    else
                     {
-                        this.sourceCode += System.IO.File.ReadAllText(file) + "\n";
+
+                        long initLength = 0;
+
+                        // We replace 'ReadAllText' with 'StringBuilder' and 'ReadAllLines'
+                        // to avoide temporary LOH allocations.
+                        foreach (string file in paths)
+                        {
+                            FileInfo f = new FileInfo(file);
+                            initLength += f.Length;
+                        }
```

Comment:
Calculate the total size might not be worth it. How about we just use a relatively large initial length? Say 8192 (8k)?
To be honest, I think if a user feed a file with more than 85k in size to Add-Type, they are using it wrongly (that Azure service might use `Add-Type` wrongly).

Reply:
Could you ask Azure profile team about this? I guess they can have real things.

Reply:
Do you mean asking about the azure service might misuse Add-Type? @PaulHigin already replied them and brought up this to them.

Reply:
I meant what is typical file size they see in their traces?

Reply:
If you mean the initial capacity for StringBuilder, I think there is no accurate answer for it as it's probably a case by case basis. The default initial capacity is 16 bytes, so I think as long as we are using a relatively large number, it should be OK. (it's micro optimization that may be premature anyway)

Reply:
I dislike always pre-allocating StringBuilder to a large size.  It may be over allocated for the majority of cases.  But if we do this we should definitely ensure we pre-allocate at a size less than a LOH block.  I like the idea of pre-allocating intelligently but this means computing the file sizes and I am concerned that may have a high cost (I seem to remember a .Net bug  where FileInfo resulted in opening files on some platforms).

I am inclined to just leave it at the .Net default allocation.  To me the big perf win was removing the duplicate file read / string allocation in the base class.

Reply:
.Net default (16 bytes) is very small for _file_ size. I think I agree that 8k is best in the case. Also we plan remove the code at all in follow PR so I ready put any value in the time :-)